include/RAJA/IndexValue.hxx

```diff
@@ -261,34 +262,35 @@ protected:
   Index_type value;
 };
 
-/*!
- * \brief Helper class for convertIndex, since functions cannot be partially
- * specialized
- */
-template <typename TO, typename FROM>
-struct ConvertIndexHelper {
-  RAJA_HOST_DEVICE
-  RAJA_INLINE
-  static TO convert(FROM val) { return TO(*val); }
-};
-
-template <typename TO>
-struct ConvertIndexHelper<TO, Index_type> {
-  RAJA_HOST_DEVICE
-  RAJA_INLINE
-  static TO convert(Index_type val) { return TO(val); }
-};
 
 /*!
  * \brief Function provides a way to take either an int or any Index<> type, and
  * convert it to another type, possibly another Index or an int.
+ *
+ * This has been refactored to use SFINAE to differentiate between IndexValue
+ * derived types an non-IndexValue types, and return the correct type
+ * automatically
  */
+
```

Comment:
There might be a case where an optimizing compiler takes care of understanding that only one of the branches will ever be taken given a passed in type, but the code would not technically be compliant to the language.